src/openforms/submissions/models/submission.py

```diff
@@ -642,3 +649,13 @@ def get_prefilled_data(self):
                 variable.key: variable.value for variable in prefill_vars
             }
         return self._prefilled_data
+
+    def get_registration_plugin(self):
+        from ..tasks.registration import get_registration_plugin
+
+        return get_registration_plugin(self)
+
+    def set_submission_reference(self):
+        from ..tasks.registration import set_submission_reference
```

Comment:
I think you could move this to `openforms.submissions.public_references` instead and contain the implementation details in there. Fine to also expose this as a public method then, but importing utilities from celery tasks is a bit of a no-go.